src/channel.c

```diff
@@ -545,19 +564,29 @@ static int is_banned(aClient *cptr, aChannel *chptr, chanMember *cm)
 
     strcpy(s, make_nick_user_host(cptr->name, cptr->user->username,
                                   cptr->user->host));
+#ifdef USER_HOSTMASKING
```

Comment:
@eaescob: no, we do it on register_user() which is called by m_nick():
#ifdef USER_HOSTMASKING
        strncpyzt(user->mhost, mask_host(user->host,0), HOSTLEN + 1);
#endif

SVSHOST isn't related to the user host-masking, it is only used when services change a user's host (and even then, SVSHOST is propagated after the NICK and before SJOIN so it will work either way)

-Kobi.